pyinstrument/__main__.py

```diff
@@ -325,7 +325,12 @@ def dash_m_callback(option: str, opt: str, value: str, parser: optparse.OptionPa
         session = profiler.stop()
 
     if options.outfile:
-        f = codecs.open(options.outfile, "w", "utf-8")
+        if should_close_f_after_writing:
+            f.close()
```

Comment:
This seems like it's in the wrong place, perhaps an accidental addition? I think these lines should be removed.

```suggestion
```

Reply:
Actually, there's something weird going on here... the outfile is opened twice. Must be an existing defect. I'll take a look.

Reply:
I've refactored this to remove the preexisting issue, I'll push the changes